libs/zems/build.zig

```diff
@@ -0,0 +1,179 @@
+    emsdk_path: []const u8,
+    emsdk_include_path: []const u8,
+
+    pub fn init(b: *std.Build.Builder) *@This() {
+        const emsdk_path = b.env_map.get("EMSDK") orelse @panic("Failed to get emscripten SDK path, have you installed & sourced the SDK?");
+        var r = b.allocator.create(EmscriptenStep) catch unreachable;
+        r.* = .{
+            .b = b,
+            .args = EmscriptenArgs.init(b.allocator),
+            .emsdk_path = emsdk_path,
+            .emsdk_include_path = b.pathJoin(&.{ emsdk_path, "upstream", "emscripten", "cache", "sysroot", "include" }),
+        };
+        return r;
+    }
+
+    pub fn link(self: *@This(), exe: *std.Build.CompileStep) void {
+        std.debug.assert(self.lib_exe == null);
+        std.debug.assert(self.link_step == null);
+        const b = self.b;
+
+        exe.addSystemIncludePath(.{ .path = self.emsdk_include_path });
+        exe.stack_protector = false;
+        exe.disable_stack_probing = true;
+        exe.linkLibC();
+
+        const emlink = b.addSystemCommand(&.{"emcc"});
+        emlink.addArtifactArg(exe);
+        for (exe.link_objects.items) |link_dependency| {
+            switch (link_dependency) {
+                .other_step => |o| emlink.addArtifactArg(o),
+                // .c_source_file => |f| emlink.addFileSourceArg(f.source), // f.args?
+                // .c_source_files => |fs| for (fs.files) |f| emlink.addArg(f), // fs.flags?
+                else => {},
+            }
+        }
+        const out_path: []const u8 = self.out_path orelse b.pathJoin(&.{ b.pathFromRoot("."), "zig-out", "web", exe.name });
+        std.fs.cwd().makePath(out_path) catch unreachable;
+        const out_file = std.mem.concat(b.allocator, u8, &.{ out_path, std.fs.path.sep_str ++ "index.html" }) catch unreachable;
+        emlink.addArgs(&.{ "-o", out_file });
+
+        if (self.args.exported_functions.items.len > 0) {
+            var s = std.mem.join(self.b.allocator, "','", self.args.exported_functions.items) catch unreachable;
+            var ss = std.fmt.allocPrint(self.b.allocator, "-sEXPORTED_FUNCTIONS=['{s}']", .{s}) catch unreachable;
+            emlink.addArg(ss);
+        }
+
+        var op_it = self.args.options.iterator();
+        while (op_it.next()) |opt| {
+            if (opt.value_ptr.*.len > 0) {
+                emlink.addArg(std.mem.join(b.allocator, "", &.{ "-s", opt.key_ptr.*, "=", opt.value_ptr.* }) catch unreachable);
+            } else {
+                emlink.addArg(std.mem.join(b.allocator, "", &.{ "-s", opt.key_ptr.* }) catch unreachable);
+            }
+        }
+
+        emlink.addArgs(self.args.other_args.items);
+
+        if (self.args.shell_file) |sf| emlink.addArgs(&.{ "--shell-file", sf });
+
+        emlink.step.dependOn(&exe.step);
+        self.link_step = emlink;
+        self.lib_exe = exe;
+    }
+};
+
+pub const EmscriptenArgs = struct {
+    source_map_base: []const u8 = "./",
+    shell_file: ?[]const u8 = null,
+
```

Comment:
Then add the arg:
```content_dir: ?[]const u8 = "content",```

I found you can use files like you would natively. For loading fonts, images, etc.